distributed/worker_state_machine.py

```diff
@@ -372,6 +425,16 @@ class ExecuteSuccessEvent(StateMachineEvent):
     type: type | None
     __slots__ = tuple(__annotations__)  # type: ignore
 
+    def log(self, *, handled: float) -> StateMachineEvent:
+        out = copy(self)
+        out.handled = handled
+        out.value = None
+        return out
+
+    def _after_from_dict(self) -> None:
+        self.value = None
+        self.type = None
```

Comment:
I guess the execution result type is discarded here because it's merely a string representation at this point and one would have to deal with serialising/unserialising types.

In any case, I think reconstructing the result of execution is non-trivial. How does this impact replayability of events on the Worker (out of interest?)

Reply:
these fields that are being discarded on a serialization round-trip should be inconsequential for the purpose of rebuilding  the state.